x-pack/plugin/eql/qa/common/src/main/java/org/elasticsearch/test/eql/BaseEqlSpecTestCase.java

```diff
@@ -168,20 +169,42 @@ public String toString() {
             }
         });
 
-        long[] expected = eventIds;
         long[] actual = extractIds(events);
-        assertArrayEquals(
-            LoggerMessageFormat.format(
-                null,
-                "unexpected result for spec[{}] [{}] -> {} vs {}",
-                name,
-                query,
-                Arrays.toString(expected),
-                Arrays.toString(actual)
-            ),
-            expected,
-            actual
-        );
+        if (eventIds.size() == 1) {
```

Comment:
Is this case singled out just to make use of `assertArrayEquals()`? Otherwise using the code in the other branch should be enough, imo. The message could be customized based on list size.

Reply:
It's just because assertArrayEquals() returns a more precise error message (ie. it also points to the exact array element that differs), that is useful in that specific case, but does not make much sense when you have multiple possible results